### ISSUE TITLE
Unable to install Ruby 2.2.1 on OS X 10.10.2 – "stack level too deep"

### ISSUE THREAD

**Issue by markbao** [wayneeseguin/rvm#3345]
rvm 1.26.10 (master)
OS X 10.10.2
Command: `rvm install 2.2.1`

Install error:

````
./tool/rbinstall.rb:728:in `block in <main>': stack level too deep (SystemStackError)
	from ./tool/rbinstall.rb:757:in `call'
	from ./tool/rbinstall.rb:757:in `block in <main>'
	from ./tool/rbinstall.rb:754:in `each'
	from ./tool/rbinstall.rb:754:in `<main>'
make: *** [do-install-nodoc] Error 1
+__rvm_make:0> return 2
````

Attempted instructions in #3259 (`rvm get head`, etc.) to no avail.

Full install.log: https://gist.github.com/markbao/07812ca13a9a7d66a05c

**Comment by olleromo** [comment]
Same thing here.

```
Error running '__rvm_make install',
showing last 15 lines of /Users/or/.rvm/log/1426612283_ruby-2.2.1/install.log
                              json 1.8.1
                              psych 2.0.8
                              rake 10.4.2
                              rdoc 4.2.0
installing bundle gems:       /Users/or/.rvm/rubies/ruby-2.2.1/lib/ruby/gems/2.2.0 (build_info, cache, doc, extensions, gems, specifications)
                              minitest-5.4.3.gem
                              power_assert-0.2.2.gem
                              test-unit-3.0.8.gem
./tool/rbinstall.rb:728:in `block in <main>': stack level too deep (SystemStackError)
	from ./tool/rbinstall.rb:757:in `call'
	from ./tool/rbinstall.rb:757:in `block in <main>'
	from ./tool/rbinstall.rb:754:in `each'
	from ./tool/rbinstall.rb:754:in `<main>'
make: *** [do-install-nodoc] Error 1
+__rvm_make:0> return 2
```

Also tried the #3259 solution.

**Comment by olleromo** [comment]
It's working now. I imploded RVM, fixed everything 'brew doctor' came up with, upgraded xcode and reinstalled command line tools and now everything works. Obviously all the things I should have done before complaining here. Thank you for RVM - it's just great :)

**Comment by mpapis** [comment]
fixed in #3355

**Comment by nickysemenza** [comment]
note that to get the changes from #3355 to work you need to do this:
`rvm get head`
`rvm reinstall ruby-2.2.1`
